src/components/link/link.helper.js

```diff
@@ -57,22 +58,60 @@ function getRadiusStrategy(type) {
     return RADIUS_STRATEGIES[type] || RADIUS_STRATEGIES[LINE_TYPES.STRAIGHT];
 }
 
+const cacheLinksCount = new Map();
+const cacheIdCount = new Map();
+const _key = (s, t) => `${s}:${t}`;
+const addLink = (s, t, id) => {
+    if (!s || !t || !id) {
+        return;
+    }
+    const k = _key(s, t);
+    let c = 0;
+    if (cacheLinksCount.has(k)) {
+        const s = cacheLinksCount.get(k);
+        s.add(id);
+        c = s.size;
+        cacheLinksCount.set(k, s);
+    } else {
+        const s = new Set();
+        s.add(id);
+        c = s.size;
+        cacheLinksCount.set(k, s);
+    }
+    if (!cacheIdCount.has(id)) {
+        cacheIdCount.set(id, c);
+    }
+};
+const getCount = id => (id ? cacheIdCount.get(id) : undefined);
+
 /**
  * This method returns the path definition for a given link base on the line type
  * and the link source and target.
  * {@link https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d|d attribute mdn}
  * @param {Object} link - the link to build the path definition
+ * @param {string} link.id - optional id that client might set for a link
  * @param {Object} link.source - link source
  * @param {Object} link.target - link target
  * @param {string} type - the link line type
  * @returns {string} the path definition for the requested link
  * @memberof Link/helper
  */
-function buildLinkPathDefinition({ source = {}, target = {} }, type = LINE_TYPES.STRAIGHT) {
+function buildLinkPathDefinition({ id, source, target }, type = LINE_TYPES.STRAIGHT) {
```

Comment:
- [ ] To avoid confusion, better rename `source` to `sourceInfo` since it's an object and not the actual source identifier